tests/src/test/scala/slinky/core/annotations/ReactAnnotatedComponentTest.scala

```diff
@@ -213,6 +215,67 @@ object DerivedStateComponent {
   }
 }
 
+@react class SubComponentWithFunctorReactElements extends StatelessComponent {
+  case class Props(seq: Seq[ReactElement],
+                   list: List[ReactElement],
+                   option: Option[ReactElement],
+                   attempt: Try[ReactElement],
+                   function: () => ReactElement,
+                   fragment: ReactElement,
+                   map: Map[String, ReactElement],
+                   tuple: (String, ReactElement),
+                   future: Future[ReactElement],
+                   varargs: ReactElement*)
+
+  override def render(): ReactElement = {
+    div(props.seq, props.list, props.option, props.attempt, props.function, props.fragment, props.varargs)
+  }
+}
+
+@react class ComponentWithVariableFunctorReactElements extends StatelessComponent {
+
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  type Props = Unit
+
+  override def render(): ReactElement = {
+    val s = Seq("a", "b")
+    val l = List("c", "d")
+    val o = Some("e")
+    val t = Try(h2("f"))
+    val fn = () => "g"
+    val fr = Fragment(List(h1("h"), i("i")))
+    val m = Map(("j", "k"))
+    val t2 = ("l", "m")
+    val f = Future { "n" }
```

Comment:
How does this work right now? If the future is executed asynchronously, how is the underlying element passed over to React?

Reply:
Most of this doesn't work right now. And you can get completely different if you use a variable or not. If you open up that test in IntelliJ and enable Show Implicit Hints you see the issue.